assembly/src/assembler/span_builder.rs

```diff
@@ -144,21 +144,30 @@ impl SpanBuilder {
     /// This consumes all operations and decorators in the builder, but does not touch the
     /// operations in the epilogue of the builder.
     pub fn extract_span_into(&mut self, target: &mut Vec<CodeBlock>) {
-        if !self.ops.is_empty() {
-            let ops = self.ops.drain(..).collect();
-            let decorators = self.decorators.drain(..).collect();
+        let ops: Vec<Operation> = self.ops.drain(..).collect();
+        let decorators: DecoratorList = self.decorators.drain(..).collect();
+
+        if !ops.is_empty() {
             target.push(CodeBlock::new_span_with_decorators(ops, decorators));
-        } else if !self.decorators.is_empty() {
-            // this is a bug in the assembler. we shouldn't have decorators added without their
-            // associated operations
-            unreachable!()
+        } else if !decorators.is_empty() {
+            // this is a bug in the assembler. we shouldn't have AsmOp decorators added without
+            // their associated operations
+            if self
+                .decorators
+                .iter()
+                .any(|(_, decorator)| matches!(decorator, Decorator::AsmOp(_)))
+            {
+                unreachable!()
+            }
+
+            target.push(CodeBlock::new_span_with_decorators(vec![Operation::Noop], decorators));
```

Comment:
Now that I think about it more, I'm not sure we should allow this. Maybe rather than panicking we should return an error somehow - but even that is not ideal.

Reply:
Or maybe returning an error is fine. A span block with decorators and no operations is most likely a mistake (or at least I can't think of why someone would do it).

Reply:
question: Isn't something like this valid?
....
....
if.true
    adv.keyval
else
....
....